sdk/monitor/Azure.Monitor.Query/tests/MonitorQueryTestEnvironment.cs

```diff
@@ -16,10 +17,32 @@ public class MonitorQueryTestEnvironment : TestEnvironment
         public string MetricsNamespace => GetRecordedVariable("METRICS_RESOURCE_NAMESPACE");
         public string MonitorIngestionEndpoint => GetOptionalVariable("METRICS_INGEST_SUFFIX") ?? "ods.opinsights.azure.com";
         public string MetricsIngestionEndpoint => GetOptionalVariable("METRICS_INGEST_SUFFIX") ?? "monitoring.azure.com";
-        public Uri LogsEndpoint => new(GetRecordedVariable("LOGS_ENDPOINT"));
         public Uri MetricsEndpoint => new(ResourceManagerUrl);
         public string ResourceId => GetRecordedVariable("RESOURCE_ID");
         public string WorkspacePrimaryResourceId => GetRecordedVariable("WORKSPACE_PRIMARY_RESOURCE_ID");
         public string WorkspaceSecondaryResourceId => GetRecordedVariable("WORKSPACE_SECONDARY_RESOURCE_ID");
+        public Uri LogsEndpoint => new(GetEndpoint());
```

Comment:
This will fail during playback as we are no longer looking up the recorded variable.

Reply:
I can check if the mode is playback and set it to the default?

Reply:
No, if it is playback it would need to check the recorded value in the session record, which is what GetRecordedVariable does. 

It would probably be cleaner if we could move the GetEndpoint logic directly into the bicep file so that we are setting the LOGS_ENDPOINT env var based on the Azure Environment.

Reply:
If we don't want to go that route, then we would at least need to condition on whether we are in PlayBack mode and if so, call GetRecordedVariable rather than GetEndpoint.